routing/routing.go

```diff
@@ -259,6 +261,19 @@ func New(o Options) *Routing {
 		o.Log = &logging.DefaultLog{}
 	}
 
+	uniqueClients := make(map[DataClient]struct{}, len(o.DataClients))
+	for i, c := range o.DataClients {
+		if _, ok := uniqueClients[c]; ok {
+			o.Log.Errorf("Duplicate data clients are not allowed, ignoring client #%d: %T", i, c)
```

Comment:
We can make it work with duplicates as well but I don't see a usecase for it - its likely a misconfiguration that currently goes unnoticed.

Reply:
This supports multiple dataclient instances of the same time but excludes equal duplicates i.e. equal pointers or value types which is very likely a misconfguration.

Reply:
Ah ok it wasn't clear to me.